From 43018b655f56087d3f5816b693d83e1c377cda45 Mon Sep 17 00:00:00 2001
From: Enzo <enzo@homelab.local>
Date: Thu, 4 Jun 2026 08:38:10 -0400
Subject: [PATCH 3/4] phy: allwinner: add A733 USB PHY compatible

---
 drivers/phy/allwinner/phy-sun4i-usb.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/phy/allwinner/phy-sun4i-usb.c b/drivers/phy/allwinner/phy-sun4i-usb.c
index e2fbf8ccf99e..92c862aecfe5 100644
--- a/drivers/phy/allwinner/phy-sun4i-usb.c
+++ b/drivers/phy/allwinner/phy-sun4i-usb.c
@@ -1028,6 +1028,7 @@ static const struct of_device_id sun4i_usb_phy_of_match[] = {
 	{ .compatible = "allwinner,sun50i-a64-usb-phy",
 	  .data = &sun50i_a64_cfg},
 	{ .compatible = "allwinner,sun50i-h6-usb-phy", .data = &sun50i_h6_cfg },
+	{ .compatible = "allwinner,sun60i-a733-usb-phy", .data = &sun50i_h616_cfg },
 	{ .compatible = "allwinner,sun50i-h616-usb-phy", .data = &sun50i_h616_cfg },
 	{ .compatible = "allwinner,suniv-f1c100s-usb-phy",
 	  .data = &suniv_f1c100s_cfg },
-- 
2.53.0

