From 92bc10d7949d0b2df030e2318023f62b916b9636 Mon Sep 17 00:00:00 2001
From: Enzo <enzo@homelab.local>
Date: Thu, 4 Jun 2026 08:42:40 -0400
Subject: [PATCH 3/3] arm64: dts: allwinner: add A733 GPADC node

---
 .../arm64/boot/dts/allwinner/sun60i-a733.dtsi | 29 +++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun60i-a733.dtsi b/arch/arm64/boot/dts/allwinner/sun60i-a733.dtsi
index 4a9a7453fe9e..dc742a39f14d 100644
--- a/arch/arm64/boot/dts/allwinner/sun60i-a733.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun60i-a733.dtsi
@@ -244,6 +244,35 @@ syscon: syscon@3000000 {
 			ranges;
 		};
 
+		gpadc: adc@2521000 {
+			compatible = "allwinner,sun60i-a733-gpadc",
+				     "allwinner,sun20i-d1-gpadc";
+			reg = <0x02521000 0x400>;
+			interrupts = <GIC_SPI 38 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&ccu CLK_BUS_GPADC>;
+			resets = <&ccu RST_BUS_GPADC>;
+			#io-channel-cells = <1>;
+
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			channel@0 {
+				reg = <0>;
+			};
+
+			channel@1 {
+				reg = <1>;
+			};
+
+			channel@2 {
+				reg = <2>;
+			};
+
+			channel@3 {
+				reg = <3>;
+			};
+		};
+
 		usb_otg: usb@4100000 {
 			compatible = "allwinner,sun8i-a33-musb";
 			reg = <0x04100000 0x400>;
-- 
2.53.0

